### PR TITLE
Use bibtool to organize bibliography

### PR DESCRIPTION
**Pull request by fingolfin**
- Add config file for [bibtool](http://www.gerd-neugebauer.de/software/TeX/BibTool/en/), available with CTAN and in many package managers
- Sort and unify the bibliography using bibtool
- Let bibtool generate (most) bib keys (one could probably fiddle with the config to do all, but that's too much work).

Now one can add new bib entries by pasting them at the end of the `.bib` file, then running this command to get them formatted and sorted:
```
bibtool docs/oscar_references.bib -o docs/oscar_references.bib
```

### CODE REVIEW

File: docs/src/Fields/intro.md [hunk 1]
```diff
@@ -24,8 +24,8 @@ various kinds of fields:
 - finite fields
 
 General textbooks offering details on theory and algorithms include:
-- [Cohen1](@cite)
-- [Cohen2](@cite)
+- [Coh93](@cite)
+- [Coh00](@cite)
 - [LN97](@cite)
-- [Marcus](@cite)
```
Comment (fingolfin) [inline docs/src/Fields/intro.md:30]: OK, I just realized these changes first also need to be made in Hecke.jl (CC @thofma). I guess until then this PR needs to be on hold anyway, but that's fine, it's not urgent :-)

Comment (ThomasBreuer) [reply docs/src/Fields/intro.md:30]: Concerning the labels, see my remark in #637

Comment (fingolfin) [reply docs/src/Fields/intro.md:30]: Ah sorry, wasn't aware. Thank you!

**Comment by fingolfin** [comment]
@thofma did you close this on purpose?

**Comment by thofma** [comment]
Have to fix one more thing.

**Comment by thofma** [comment]
Good to go. Shall I squash it into one commit?

**Pull request merged**